libopenarc/arc.h

```diff
@@ -442,6 +443,20 @@ extern ARC_STAT arc_body __P((ARC_MESSAGE *msg, u_char *buf, size_t len));
 extern ARC_STAT arc_eom __P((ARC_MESSAGE *));
 
 /*
+**  ARC_CHECK_WHITELIST -- verify that all AS d= domains are present in the provided whitelist
+**
+**  Parameters:
+**  	msg -- ARC_MESSAGE object
+**    domain_whitelist -- an array of domain pointers
+**    whitelist_size -- number of domains in the whitelist
+**  Return value:
+**  	An ARC_STAT_* constant.
+**
+*/
+
+extern ARC_STAT arc_check_whitelist (ARC_MESSAGE *, char **, u_int);
```

Comment:
There's a lot of complexity that I'm having a hard time resolving wrt to this, maybe you can provide some advice.  This code calls arc_param_get, which is defined in arc.c, but not exported in the header for public consumption.  It's prototype is:

static u_char *
arc_param_get(ARC_KVSET *set, u_char *param)
Which should probably just get added to arc.h

That ARC_KVSET is problamatic.  It's defined in arc-internal.h, which imports arc.h, so it is defined after that prototype.  Any suggestions on how to handle this?